c7n/resources/cw.py

```diff
@@ -114,6 +119,171 @@ def get_dimensions(self, resource):
+
+    .. code-block:: yaml
+
+        policies:
+            - name: find-event-rules-with-no-targets
+              resource: aws.event-rule
+              filters:
+                - type: event-rule-target
+                  key: Arn
+                  value: absent
+    """
+
+    RelatedResource = "c7n.resources.cw.EventRuleTarget"
+    RelatedIdsExpression = 'Name'
+    AnnotationKey = "EventRuleTargets"
+
+    schema = type_schema('event-rule-target', rinherit=ValueFilter.schema)
+    permissions = ('events:ListTargetsByRule',)
+
+
+@EventRule.filter_registry.register('invalid-targets')
+class ValidEventRuleTargetFilter(ChildResourceFilter):
+    """
+    Filter event rules for invalid targets, Use the `all` option to
+    find any event rules that have all invalid targets, otherwise
+    defaults to filtering any event rule with at least one invalid
+    target.
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: find-event-rules-with-invalid-targets
+              resource: aws.event-rule
+              filters:
+                - type: invalid-targets
+                  all: true # defaults to false
+    """
+
+    RelatedResource = "c7n.resources.cw.EventRuleTarget"
+    RelatedIdsExpression = 'Name'
+    AnnotationKey = "EventRuleTargets"
+
+    schema = type_schema(
+        'invalid-targets',
+        **{
+            'all': {
+                'type': 'boolean',
+                'default': False
+            }
+        }
+    )
+
+    permissions = ('events:ListTargetsByRule',)
+
+    def validate(self):
+        pass
+
+    def get_rules_with_children(self, resources):
+        """
+        Augments resources by adding the c7n:ChildArns to the resource dict
+        """
+
+        results = []
+
+        # returns a map of {parent_reosurce_id: [{child_resource}, {child_resource2}, etc.]}
+        child_resources = self.get_related(resources)
+
+        # maps resources by their name to their data
+        for r in resources:
+            if child_resources.get(r['Name']):
+                for c in child_resources[r['Name']]:
+                    r.setdefault('c7n:ChildArns', []).append(c['Arn'])
+                results.append(r)
+        return results
+
+    def process(self, resources, event=None):
+        # Due to lazy loading of resources, we need to explicilty load the following
+        # potential targets for a event rule target:
+        load_resources(
+            [
+                "aws.sqs",
+                "aws.lambda",
+                "aws.ecs-cluster",
+                "aws.ecs-task",
+                "aws.kinesis",
+                "aws.sns",
+                "aws.ssm-parameter",
+                "aws.batch-compute",
+                "aws.codepipeline",
+            ]
+        )
+        arn_resolver = ArnResolver(self.manager)
+        resources = self.get_rules_with_children(resources)
+        results = []
+
+        if self.data.get('all'):
+            op = any
```

Comment:
these operators seem reversed

Reply:
It's a bit unintuitive because we're dealing with a negative here, all invalid targets means that the variable `resolved` contains all false values which can be evaluated as 

```python
not any(resolved.values())
```

In the case that we're looking for any unresolved values, we can evaluate it like:

```python
not all(resolved.values())
```